pints/tests/test_toy_stochastic_degradation_model.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+#
+# Tests if the stochastic degradation (toy) model works.
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+import unittest
+import numpy as np
+import pints
+import pints.toy
+
+
+class TestStochasticDegradation(unittest.TestCase):
+    """
+    Tests if the stochastic degradation (toy) model works.
+    """
+
+    def test_start_with_zero(self):
+        # Test the special case where the initial concentration is zero
+        from pints.toy import StochasticDegradationModel
+        model = StochasticDegradationModel(0)
+        times = [0, 1, 2, 100, 1000]
+        parameters = [0.1]
+        values = model.simulate(parameters, times)
+        self.assertEqual(len(values), len(times))
+        self.assertTrue(np.all(values == np.zeros(5)))
+
+    def test_start_with_twenty(self):
+        # Run small simulation
+        model = pints.toy.StochasticDegradationModel(20)
+        times = [0, 1, 2, 100, 1000]
+        parameters = [0.1]
+        values = model.simulate(parameters, times)
+        self.assertEqual(len(values), len(times))
+        self.assertEqual(values[0], 20)
+        self.assertEqual(values[-1], 0)
+
+    def test_suggested(self):
+        model = pints.toy.StochasticDegradationModel(20)
+        times = model.suggested_times()
+        parameters = model.suggested_parameters()
+        self.assertTrue(np.all(times == np.linspace(0, 100, 101)))
```

Comment:
I think value based tests are overkill here. The model returns some suggested values, what they are is determined in the model, we don't need to re-specify that here.
Instead, just check if they have the correct shape and/or type.
A very easy way to do this is to use *pass them back into the model*. That's what we want really, that `suggested_parameters()` returns something - whatever that is - that can be used as parameters.
So just run a simulation with the suggested times and parameters, if that works all is fine, if that raises an exception the test will fail.

Reply:
Done